doc/threat_model/README.md

```diff
@@ -0,0 +1,173 @@
+Tock Threat Model
+=================
+
+
+**Note: This threat model is not descriptive of Tock's current implementation.
+It describes how we intend Tock to work as of some future release, perhaps
+2.0.**
+
+## Overview
+
+Tock provides hardware-based isolation between processes as well as
+language-based isolation between kernel capsules.
+
+Tock supports a variety of hardware, including boards defined in the Tock
+repository and boards defined "out of tree" in a separate repository.
+Additionally, Tock's installation model may vary between different use cases
+even when those use cases are based on the same hardware. As a result of Tock's
+flexibility, the mechanisms it uses to provide isolation — and the strength of
+that isolation — vary from deployment to deployment.
+
+This threat model describes the isolation provided by Tock as well as the trust
+model that Tock uses to implement that isolation. Users of Tock, which include
+board integrators and application developers, should use this threat model to
+understand what isolation Tock provides to them (and what isolation it may not
+provide). Tock developers should use this threat model as a guide for how to
+provide Tock's isolation guarantees.
+
+## Definitions
+
+These definitions are shared between the documents in this directory.
+
+A **process** is a runtime instantiation of an application binary. When an
+application binary "restarts", its process is terminated and a new process is
+started using the same binary. Note that the kernel is not considered a process,
+although it is a thread of execution.
+
+**Process data** includes a process' binary in non-volatile storage, its memory
+footprint in RAM, and any data that conceptually belongs to the process that is
+held by the kernel or other processes. For example, if a process requests
+samples from an ADC then those samples are considered the process' data, even
+when they are stored in a location in RAM only readable by the kernel.
```

Comment:
This line feels to me like it conflicts with  the example from Virtualization.md

> An Analog-to-Digital Converter (ADC) does not have a concept of which process "owns" data, nor is there a way to implement such a concept. As such, an ADC API that allows clients to take samples upon request does not need to take separate samples for different clients. An ADC API that receives simultaneous requests to sample the same source may take a single reading and distribute it to multiple clients.

Reply:
I agree. I changed the README example to a UART, and clarified that the Virtualization example is referring to ADC hardware. I don't think ADC samples have a clear owner.

Reply:
Thank you. This resolves my concern.